### ISSUE TITLE
On Raft leader evacuation or cert. rotation config-tx forwarding may fail silently

### ISSUE THREAD

**Issue by tock-ibm** [hyperledger/fabric#4066]
Analysis of recent flakes in Raft integration tests reveals a that the root cause (of some of them, at least) is from the leader abdication mechanism, following leader evacuation of cert rotation. Here is a the story. The test is: (config_test.go:490)
```
	When("the orderer certificates are all rotated", func() {
		It("is possible to rotate certificate by adding & removing cert in single config", func() {
```

- The test is rotating the certs of 3 node, starting from the leader
- The first leader is 2, its rotation goes fine. It detects the rotation, abdicates and transfers leadership to node 3, gets the config, and becomes unavailable, restarted, joins the cluster
- The next node to be rotated is node 1, a follower, it goes smooth. 
- The last node to be rotated is node 3, the leader, and here it fails.
- It detects the rotation, abdicates and transfers leadership to node 2
- It then tries to forward the config-tx to node 2, but fails. It does not retry.
- The config-tx never gets submitted and committed, so the test waits for node 3 to become unavailable until it times out.

The forwarding failure message is:
```
2023-02-23T14:52:51.7172154Z     [91m[e][35m[OrdererOrg.orderer3][0m [34m2023-02-23 14:51:50.363 UTC 008b INFO[0m [orderer.consensus.etcdraft] [34;1mforwardToLeader[0m -> Forwarding transaction to the leader 2 channel=testchannel node=3
2023-02-23T14:52:51.7212097Z     [91m[e][35m[OrdererOrg.orderer3][0m [34m2023-02-23 14:51:50.363 UTC 008c INFO[0m [orderer.consensus.etcdraft] [34;1mfunc1[0m -> Un-mapping transaction stream to 2 because encountered a stale stream
2023-02-23T14:52:51.7213757Z     [91m[e][35m[OrdererOrg.orderer3][0m [33m2023-02-23 14:51:50.363 UTC 008d WARN[0m [orderer.common.cluster.step] [33;1msendMessage[0m -> Send of SubmitRequest for channel testchannel with payload of size 30208 to orderer2.example.com(127.0.0.1:28008) took 128.7µs but failed due to EOF
2023-02-23T14:52:51.7215350Z     [91m[e][35m[OrdererOrg.orderer3][0m [33m2023-02-23 14:51:50.363 UTC 008e WARN[0m [orderer.consensus.etcdraft] [33;1mfunc1[0m -> Reconfiguration transaction forwarding failed with error: EOF channel=testchannel node=3
```

**Bottom line**: forwarding the config-tx after abdication may fail silently and sporadically. When that happens, tests that expect the config-tx to commit get stuck.

Two things may be done.

1. One option is to do some retries here (chain.go L962): 
```
// Else, abdication succeeded, so we submit the transaction (which forwards to the leader)
						if err := c.Submit(msg, 0); err != nil {
							c.logger.Warnf("Reconfiguration transaction forwarding failed with error: %v", err)
						}
						return
```
the connection will be re-established and the message will be sent.

2. Another option is to rewrite the tests such that leader evacuation /cert  rotation are retried if they get stuck.

This issue will address the first option.

**Comment by tock-ibm** [comment]
See also #4063

**Comment by tock-ibm** [comment]
https://github.com/hyperledger/fabric/pull/4024 addressed here
